tests/rest/admin/test_room.py

```diff
@@ -721,14 +736,20 @@ def test_delete_same_room_twice(self) -> None:
 
         body = {"new_room_user_id": self.admin_user}
 
+        # Mock PaginationHandler.purge_room to sleep for 100s, so we have time to do a second call
+        # before the purge is over. Note that it doesn't purge anymore, but we don't care.
+        async def purge_room(room_id: str, force: bool) -> None:
+            await deferLater(self.hs.get_reactor(), 100, lambda: None)
+
+        self.pagination_handler.purge_room = AsyncMock(side_effect=purge_room)  # type: ignore[method-assign]
```

Comment:
Otherwise the first purge would have completed before we do the second one so it would not be rejected.
With `await_result=False`, the second call is done before the info reaches the DB so it was also not being rejected.

That is my solution to that, not sure if we can do better.